tensorflow/compiler/mlir/xla/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -31,6 +31,61 @@ namespace mlir {
 namespace xla_lhlo {
 namespace {
 
+template<typename LhloOpTy>
+struct DotOpConverter : public OpRewritePattern<LhloOpTy>{
+  using OpRewritePattern<LhloOpTy>::OpRewritePattern;
+
+// Only Rank 2 tensors
+
+  LogicalResult matchAndRewrite(LhloOpTy op,
+                                PatternRewriter& rewriter) const override {
+
+    const auto& lhs = op.lhs();
+    const auto& rhs = op.lhs();
+    const auto& lhs_type = lhs.getType().template cast<MemRefType>();
+    const auto& rhs_type = rhs.getType().template cast<MemRefType>();
+    const auto& element_type = lhs_type.getElementType();
+    const auto& shape_lhs = lhs_type.getShape();
+    const auto& shape_rhs = rhs_type.getShape();
+    if (shape_lhs[shape_lhs.size()-1] != shape_rhs[shape_rhs.size()-1]) {
```

Comment:
This will crash for 1-d tensors. Please check pre condition and bail out if it doesn't fit what you handle. You can't crash for any input.